src/pytorch_lightning/trainer/supporters.py

```diff
@@ -52,24 +53,24 @@ class TensorRunningAccum:
     """
 
     def __init__(self, window_length: int):
-        self.window_length = window_length
-        self.memory = None
-        self.current_idx: int = 0
-        self.last_idx: Optional[int] = None
-        self.rotated: bool = False
+        self.reset(window_length)
 
     def reset(self, window_length: Optional[int] = None) -> None:
         """Empty the accumulator."""
-        if window_length is None:
-            window_length = self.window_length
-        self.__init__(window_length)
+        if window_length is not None:
+            self.window_length = window_length
```

Comment:
`self.window_length` should be defined when this is False too

Reply:
Not sure how can we do this cleanly. Maybe add `self.window_length = window_length` in `__init__` to make sure that `self.window_length` is always defined?

Reply:
Yes